activestorage/app/jobs/active_storage/create_variants_job.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class ActiveStorage::CreateVariantsJob < ActiveStorage::BaseJob
```

Comment:
The `CreateVariantsjob` is responsible for:
1. Determining when a preview is necessary for creating variants and generating it.
2. Either scheduling `TransformJob`s for each transformation or executing them immediately, depending on the generation strategy.